### DISCUSSION TITLE
How can i reliably pull heart-rate data from mi-band 7 to display in a game?

### DISCUSSION THREAD

**Discussion by DarkVader135** [zepp-health/zeppos-forum#135] [Q&A] [answered]
so what i am trying to achieve here is build an web-app using Huami rest-api and have that app communicate over [OSC](https://en.wikipedia.org/wiki/Open_Sound_Control) and have heart-rate data sent over local-host:9000 to the game for example [VRChat](https://youtu.be/OHjN_q6RqGY) how would i approach this? (in a more generalized structure)

==some other questions i have==
-how fast can i pull data from Huami api?
-how many times per-second can i pull heart rate data?

==other notes==
-i am a bit new to web-development and still learning [this](https://darkvader135.github.io/website-project/) was one of my first websites tho i am willing to do the effort all i am asking is for some tips to point me to the right direction.

**Comment by MyoungXUE** [comment, marked as answer]
Hello, unfortunately this feature is not available on mi-band 7 because it cannot support Side App!

In addition, I have some information to share with you.

The huami rest-api is currently only available for business partnerships and not for individual developers.

Zepp OS devices, with the exception of the mi-band 7, can connect to the network and communicate with servers in real time via Side App.

The data values obtained from the continuous measurement of the heart rate sensor are updated approximately every second.

**Reply by DarkVader135** [reply]
thank you a lot this was very helpful
